contracts/exchange-wrappers/Starkware.sol

```diff
@@ -35,6 +61,56 @@ contract Starkware is I_StarkwareContract {
         uint256 quantizedAmount
     ) external override
     {
+        uint256 amount = quantizedAmount;
+
+        address tokenAddress = extractContractAddress(assetType);
+        IERC20 token = IERC20(tokenAddress);
+        uint256 exchangeBalanceBefore = token.balanceOf(address(this));
+        bytes memory callData = abi.encodeWithSelector(
+            token.transferFrom.selector,
+            msg.sender,
+            address(this),
+            amount
+        );
+        safeTokenContractCall(tokenAddress, callData);
+
+        uint256 exchangeBalanceAfter = token.balanceOf(address(this));
+        require(exchangeBalanceAfter >= exchangeBalanceBefore, "OVERFLOW");
```

Comment:
I don't think overflow with contract balance is possible, because it would require the max supply of a token to be greater than the max uint256 value.

Reply:
Im just following the starkware contract for this mock